### PR TITLE
[C-608] Fix tooltip bugs

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Fix tooltip bugs. There were kind of a few things that were causing jankiness, primarily the timeout though, which is a blunt weapon to handle this kinda stuff. This is why I kinda dislike antd... Have to learn about the internals a little bit to get the behavior.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Focused on 3 places
- GiantTrackTile (track page): to test the bug in the linear issue -- tooltip appearing after clicking
- Receive $AUDIO modal -- to test that the tooltip goes away when you click, even if you're still hovering over the element
- Track tile on feed: to test that the tooltip goes away when you click, but also that you can summon it by hovering over again

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-c-608

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-c-608

File: packages/web/src/components/tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -23,16 +23,38 @@ export const Tooltip = ({
...
+  const onClick = useCallback(() => {
+    if (shouldDismissOnClick) {
+      setIsHidden(true)
     }
-  }, [hideTooltip])
+  }, [shouldDismissOnClick, setIsHidden])
+
+  const onMouseOut = useCallback(() => {
+    mousedOver.current = false
+    setIsHidden(false)
```
Comment (Kyle-Shanks) [inline packages/web/src/components/tooltip/Tooltip.tsx:41]: should this be hiding the tooltip or un-hiding it?

Comment (raymondjacobson) [reply packages/web/src/components/tooltip/Tooltip.tsx:41]: this is setting the overriding hidden state to false, which is different from actually hiding it. I can rename the state var to make it more understandable

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-c-608

**Review by Kyle-Shanks — APPROVED** [review]

**Pull request merged**